libs/sdk-ui-kit/src/Dialog/StylingEditorDialog/StylingEditorDialog.tsx

```diff
@@ -0,0 +1,160 @@
+// (C) 2022 GoodData Corporation
+import React, { useMemo, useState } from "react";
+import { IDialogBaseProps } from "../typings";
+import { Dialog } from "../Dialog";
+import { Typography } from "../../Typography";
+import cx from "classnames";
+import { StylingExample } from "./StylingExample";
+import { IThemeMetadataObject } from "@gooddata/sdk-model";
+import { BubbleHeaderSeparator } from "./BubbleHeaderSeparator";
+import { StylingEditorDialogFooter, IStylingEditorDialogFooterProps } from "./StylingEditorDialogFooter";
+import { IntlWrapper } from "@gooddata/sdk-ui";
+import { useIntl } from "react-intl";
+import { Message } from "../../Messages";
+
+/**
+ * @internal
+ */
+export interface IStylingEditorDialogProps extends IDialogBaseProps, IStylingEditorDialogFooterProps {
+    title: string;
+    tooltip?: string;
+    stylingContent?: IThemeMetadataObject;
+    examples?: IThemeMetadataObject[];
+    exampleToColorPreview?: (example: IThemeMetadataObject) => string[];
+    locale?: string;
+}
+
+/**
+ * @internal
+ */
+export const StylingEditorDialog = (props: IStylingEditorDialogProps) => {
+    return (
+        <IntlWrapper locale={props.locale}>
+            <StylingEditorDialogCore {...props} />
+        </IntlWrapper>
+    );
+};
+
+const StylingEditorDialogCore = (props: IStylingEditorDialogProps) => {
+    const {
+        title,
+        tooltip,
+        link,
+        stylingContent,
+        examples,
+        exampleToColorPreview,
+        onClose,
+        onSubmit,
+        onCancel,
+    } = props;
+    const providedExamples = !!examples && examples.length !== 0 && !!exampleToColorPreview;
+    const [nameField, setNameField] = useState(stylingContent?.title ?? "");
+    const [definitionField, setDefinitionField] = useState(
+        stylingContent?.theme ? JSON.stringify(stylingContent?.theme, null, 4) : "",
+    );
+    const [invalidDefinition, setInvalidDefinition] = useState(false);
+    const isSubmitDisabled = useMemo(
+        () => nameField === "" || definitionField === "",
+        [nameField, definitionField],
+    );
+    const intl = useIntl();
+    const validateDefinition = (value: string): boolean => {
+        try {
+            JSON.parse(value);
+            return true;
+        } catch (e) {
+            setInvalidDefinition(true);
```

Comment:
I would rather set this flag on single place in handleDefinitionFieldChange. Now half of cases is treated here and half there